workflows/sarek.nf

```diff
@@ -1005,6 +1005,29 @@ def extract_csv(csv_file) {
         }
     }
 
+    // check that the sample sheet doesn't contain
+    // 1. multiple rows with the same combination of patient, sample and lane, and
+    // 2. same sample for different patients 
+    def patient_sample_lane_combinations_in_samplesheet = []
+    def sample2patient = [:]
+
+    Channel.from(csv_file).splitCsv(header: true)
+        .map{ row ->
+            def patient_sample_lane = [row.patient.toString(), row.sample.toString(), row.lane.toString()]
```

Comment:
You should check if `row.lane` exists

Reply:
I could do that, but, as far as I can see, it is not necessary. If the lane-field is empty, then the pipeline will stop with the error message:
```
WARN  nextflow.Nextflow - Missing or unknown field in csv file header. Please check your samplesheet
```
More details here:  https://github.com/nf-core/sarek/issues/93#issuecomment-1176186053

Reply:
Okay, I indeed did my test with `--step mapping`.

I'm not sure what you mean by "you should check if `row.lane` exists".

Do you mean that if the lane-column is present in the csv-file, then (1) applies, and otherwise then (1b) the sample sheet should not contain several rows with the same combination of patient and sample?

Reply:
I did another test of the case where the lane-column isn't present in the input-csv-file. I'm running dev-sarek like this:

```nextflow run main.nf -profile singularity --skip_tools multiqc --input samples.csv  --vep_cache_version 104 --igenomes_base /faststorage/home/aspe/igenomes```
that is, I didn't set `--step` to `mapping`. This was the sample-sheet:
```
$ cat samples.csv
patient,sex,sample,fastq_1,fastq_2
NA12878_1,XX,NA12878_1,NA12878-DNA_Blood-WGS_v2_SUB_R1.fastq.gz,NA12878-DNA_Blood-WGS_v2_SUB_R2.fastq.gz
NA12878_2,XX,NA12878_2,NA12878-DNA_Blood-WGS_v2_SUB_R1.fastq.gz,NA12878-DNA_Blood-WGS_v2_SUB_R2.fastq.gz
```
that is, a csv-file without the lane-column. The pipeline stops with message
```
WARN  nextflow.Nextflow - Missing or unknown field in csv file header. Please check your samplesheet
```
Doesn't this contradict what you mentioned above about `lane` being an optional field?

As expected, the following run:

`nextflow run main.nf -profile singularity --skip_tools multiqc --input samples_with_lane.csv  --vep_cache_version 104 --igenomes_base /faststorage/home/aspe/igenomes
`
with the following sample-sheet (with lane-field):
`
$ cat samples_with_lane.csv
patient,sex,sample,lane,fastq_1,fastq_2
NA12878_1,XX,NA12878_1,L1,NA12878-DNA_Blood-WGS_v2_SUB_R1.fastq.gz,NA12878-DNA_Blood-WGS_v2_SUB_R2.fastq.gz
NA12878_2,XX,NA12878_2,L2,NA12878-DNA_Blood-WGS_v2_SUB_R1.fastq.gz,NA12878-DNA_Blood-WGS_v2_SUB_R2.fastq.gz
`
runs without problems.

Reply:
Default step is mapping

Reply:
> Default step is mapping

Great. How do I get Sarek to do something sensible other than the mapping?

Reply:
Oooh 
Then let's try instead just is step is different than mapping then skip the first check

Reply:
Okay, I changed the code so that the first check is just applied when step=mapping.

For step=mapping, I also added an explicit check for presence of lane-data. The check is not strictly necessary, since the pipeline would fail anyway with the generic error message 
```
WARN  nextflow.Nextflow - Missing or unknown field in csv file header. Please check your samplesheet".
```
However, with the explicit check, I think the "idea" become a little clearer and a more informative error message is provided.

Reply:
Perfect